sqlglot/parser.py

```diff
@@ -1099,8 +1099,33 @@ class Parser(metaclass=_Parser):
         "WITH": lambda self: self.expression(
             exp.Properties, expressions=self._parse_wrapped_properties()
         ),
+        "BUCKET": lambda self: self._parse_partitioned_by_bucket_or_truncate(),
+        "TRUNCATE": lambda self: self._parse_partitioned_by_bucket_or_truncate(),
     }
 
+    def _parse_partitioned_by_bucket_or_truncate(self) -> exp.Expression:
+        klass = (
+            exp.PartitionedByBucket
+            if self._prev.text.upper() == "BUCKET"
+            else exp.PartitionByTruncate
+        )
+
+        this, expression = self._parse_wrapped_csv(
+            lambda: self._parse_primary() or self._parse_column()
+        )
```

Comment:
I'd implement this a little more defensively to avoid "not enough elements to expand" errors:

```python
args = self._parse_wrapped_csv(lambda: self._parse_primary() or self._parse_column())
this, expression = seq_get(args, 0), seq_get(args, 1)
```